src/modules/Packages.rb

```diff
@@ -731,19 +733,28 @@ def product_update_summary(products)
       end
 
       ret += status[:removed].map do |product|
-        transact_by = product["transact_by"]
-        log.warn "Product will be removed (by #{transact_by}): #{product}"
+        obsolete = Y2Packager::ProductUpgrade.will_be_obsoleted_by(product["name"])
+        if obsolete.empty?
+          transact_by = product["transact_by"]
+          log.warn "Product will be removed (by #{transact_by}): #{product}"
+
+          # Removing another product might be an issue
+          # (just warn if removed by user or by YaST)
+          msg = if [:user, :app_high].include?(transact_by)
+            _("<b>Warning:</b> Product <b>%s</b> will be removed.") % h(product_label(product))
+          else
+            _("<b>Error:</b> Product <b>%s</b> will be automatically removed.") %
+              h(product_label(product))
+          end
 
-        # Removing another product might be an issue
-        # (just warn if removed by user or by YaST)
-        msg = if [:user, :app_high].include?(transact_by)
-          _("<b>Warning:</b> Product <b>%s</b> will be removed.") % h(product_label(product))
+          re = HTML.Colorize(msg, "red")
         else
-          _("<b>Error:</b> Product <b>%s</b> will be automatically removed.") %
-            h(product_label(product))
+          # TRANSLATORS: Old product %{old_product} will be obsoleted by %{new_product} products.
+          _("Product <b>%{old_product}</b> will be updated to <b>%{new_product}</b>") % {
```

Comment:
um, should it start with `re = _("Product ...` ? Otherwise the `re` value will be `nil` and be returned below...